haigha/reader.py

```diff
@@ -232,7 +232,7 @@ def read_timestamp(self):
         Will raise BufferUnderflow if there's not enough bytes in the buffer.
         Will raise struct.error if the data is malformed
         """
-        return datetime.utcfromtimestamp(self.read_longlong())
+        return datetime.utcfromtimestamp(self.read_longlong() / 1e6)
```

Comment:
I saw this change and didn't (don't) have time to think through it, but I don't like the idea of changing the protocol implementation to match some non-broker spec. `time_t`, which the AMQP spec uses as a reference, is in seconds.